database/migrations/2024_08_23_165504_create_semesters_table.php

```diff
@@ -16,10 +17,10 @@ public function up()
         Schema::create('semesters', function (Blueprint $table) {
             $table->id();
             $table->string('title', 255);
-            $table->integer('version');
-            $table->enum('academic_program', array_keys(\App\Domains\Semester\Models\Semester::ACADEMIC_PROGRAMS));
+            $table->enum('version', array_keys(Semester::getVersions()));
+            $table->enum('academic_program', array_keys(Semester::getAcademicPrograms()));
             $table->text('description')->nullable();
-            $table->string('url', 255)->unique();
+            $table->string('url', 191)->unique();
```

Comment:
Any special reason to make this 191 ?

Reply:
My database didn't allow that. I found that varchar(255) field length with 'utf8mb4' can exceed 1000 byte limit for unique indexes. so I reduced the character size.